AirCasting/SessionsSynchronization/Controller/SessionSynchronizationController.swift

```diff
@@ -36,12 +38,16 @@ final class SessionSynchronizationController: SessionSynchronizer {
     func triggerSynchronization(completion: (() -> Void)?) {
         lock.lock(); defer { lock.unlock() }
         if syncInProgress { return }
-        syncInProgress = true
+        DispatchQueue.main.async {
```

Comment:
this flag is guarding us against nasty race condition bugs, it needs to be set here in a synchronous manner. It's not designed to be a UI-driver. I would suggest leaving the class as it was and build a ViewModel on top instead.

Reply:
this issue stands